### PR TITLE
Cleaned up everything that isn't needed for PHP 7.1

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | -
| Tests pass?   | ?
| Fixed issues  | None but is related to #11397

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,8 +21,7 @@ The framework is easy to adjust to meet your needs, because Yii has been designe
 Installation
 ------------
 
-- The minimum required PHP version of Yii is PHP 5.4.
-- It works best with PHP 7.
+- The minimum required PHP version of Yii is PHP 7.1.
```
Comment (cebe) [inline README.md]: should probably state the version: `The minimum required PHP version of Yii 2.1 is PHP 7.1.`

File: docs/guide/intro-upgrade-from-v1.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ the [Creating Extensions](structure-extensions.md#creating-extensions) section o
 PHP Requirements
 ----------------
 
-Yii 2.0 requires PHP 5.4 or above, which is a huge improvement over PHP version 5.2 that is required by Yii 1.1.
+Yii 2.0 requires PHP 7.1 or above, which is a huge improvement over PHP version 5.2 that is required by Yii 1.1.
```
Comment (cebe) [inline docs/guide/intro-upgrade-from-v1.md]: This should stay as is here, we should add a upgrade from 2.0 section.

File: docs/guide/intro-yii.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ This guide is mainly about version 2.0.
 Requirements and Prerequisites
 ------------------------------
 
-Yii 2.0 requires PHP 5.4.0 or above and runs best with the latest version of PHP 7. You can find more detailed
+Yii 2.0 requires PHP 7.1 or above. You can find more detailed
```
Comment (cebe) [inline docs/guide/intro-yii.md]: 2.0 -> 2.1

File: docs/guide/tutorial-yii-integration.md [hunk 4]
```diff
@@ -125,7 +125,7 @@ If you were using Yii 1 previously, it is likely you have a running Yii 1 applic
 the whole application in Yii 2, you may just want to enhance it using some of the features only available in Yii 2.
 This can be achieved as described below.
 
-> Note: Yii 2 requires PHP 5.4 or above. You should make sure that both your server and the existing application
+> Note: Yii 2 requires PHP 7.1 or above. You should make sure that both your server and the existing application
```
Comment (cebe) [inline docs/guide/tutorial-yii-integration.md]: also state the concrete version, "Yii 2.1 requires"

File: framework/base/Security.php [hunk 5]
```diff
@@ -605,19 +470,8 @@ public function generatePasswordHash($password, $cost = null)
...
-
-        $salt = $this->generateSalt($cost);
-        $hash = crypt($password, $salt);
-        // strlen() is safe since crypt() returns only ascii
-        if (!is_string($hash) || strlen($hash) !== 60) {
-            throw new Exception('Unknown error occurred while generating hash.');
-        }
-
-        return $hash;
+        /* @noinspection PhpUndefinedConstantInspection */
```
Comment (cebe) [inline framework/base/Security.php]: inspection annotation is not needed as the constant should be there in any version of PHP >=7.1

File: framework/base/Security.php [hunk 6]
```diff
@@ -642,66 +496,19 @@ public function validatePassword($password, $hash)
             throw new InvalidArgumentException('Hash is invalid.');
```
Comment (cebe) [inline framework/base/Security.php]: is this check needed? no need to verify PHP internals, `password_verify` should do that for us.

File: tests/framework/base/SecurityTest.php [hunk 7]
```diff
@@ -1219,43 +1219,6 @@ public function testHkdf($hash, $ikm, $salt, $info, $l, $prk, $okm)
...
-
-    /**
-     * @dataProvider dataProviderCompareStrings
-     *
-     * @param $expected
-     * @param $actual
-     */
-    public function testCompareStrings($expected, $actual)
-    {
-        $this->assertEquals(strcmp($expected, $actual) === 0, $this->security->compareString($expected, $actual));
```
Comment (cebe) [inline tests/framework/base/SecurityTest.php]: why did you remove this?

File: framework/UPGRADE.md [hunk 8]
```diff
@@ -61,8 +61,9 @@ Upgrade from Yii 2.0.x
 * `::className()` method calls should be replaced with native `::class`.
 * XCache and Zend data cache support was removed. Switch to another caching backends.
 * Rename `InvalidParamException` usage to `InvalidArgumentException`.
-* Moved masked input field widget was moved into separate extension https://github.com/yiisoft/yii2-maskedinput.
+* Masked input field widget was moved into separate extension https://github.com/yiisoft/yii2-maskedinput.
   Include it in your composer.json if you use it.
+* If you're used `Security::compareString()`, replace it with native `hash_equals()`.
```
Comment (francislavoie) [inline framework/UPGRADE.md]: Bad grammar here, maybe "If you're using"?

File: tests/framework/base/SecurityTest.php (hunk 7 above)
Comment (samdark) [inline tests/framework/base/SecurityTest.php]: Because there's no method anymore. There's a native one in PHP 7.

Comment (cebe) [inline tests/framework/base/SecurityTest.php]: its still there: https://github.com/yiisoft/yii2/pull/14511/files#diff-45220ab88012b2d4eb27a69410afb5e6R501

Comment (samdark) [inline tests/framework/base/SecurityTest.php]: Oh... right. I'm not sure if it's better to keep it or remove it...

Comment (cebe) [inline tests/framework/base/SecurityTest.php]: I'd keep it to provide complete interface, we also have password hash method that just wraps PHP now.

Comment (samdark) [inline tests/framework/base/SecurityTest.php]: Makes sense.
